tools/build_defs/repo/http.bzl

```diff
@@ -55,25 +55,29 @@ def _http_archive_impl(ctx):
     patch(ctx)
     workspace_and_buildfile(ctx)
 
+_HTTP_FILE_FORBIDDEN_NAMES = ["WORKSPACE", "BUILD", "BUILD.bazel"]
+
 _HTTP_FILE_BUILD = """
 package(default_visibility = ["//visibility:public"])
-
 filegroup(
     name = "file",
-    srcs = ["downloaded"],
+    srcs = ["%s"],
 )
 """
 
 def _http_file_impl(ctx):
     """Implementation of the http_file rule."""
+    if ctx.attr.downloaded_file_name in _HTTP_FILE_FORBIDDEN_NAMES:
```

Comment:
This check is not enough. What about `file`, `./WORKSPACE`,  special names like `.`, or path traversals like `../../../../.bashrc`?

Reply:
changed strategy to verify invalid files, let me know if this looks better.